```
/home/runner/work/Cataclysm-DDA/Cataclysm-DDA/Cataclysm-DDA/src/horde_map.h:1:1: error: add '#include <optional>' (for std::optional)
/home/runner/work/Cataclysm-DDA/Cataclysm-DDA/Cataclysm-DDA/src/horde_map.h:1:1: error: add '#include <optional>' (for std::optional)
```

--- a/src/horde_map.cpp
+++ b/src/horde_map.cpp
@@ -1,6 +1,7 @@
 #include "horde_map.h"
 
 #include <memory>
+#include <optional>
 #include <string>
 #include <tuple>
 
@@ -130,10 +131,10 @@ static bool is_alert( const mtype &type )
 }
 
 // These have no goal so they can't go in the active map.
-std::unordered_map<tripoint_abs_ms, horde_entity>::iterator horde_map::spawn_entity(
+std::optional<std::unordered_map<tripoint_abs_ms, horde_entity>::iterator> horde_map::spawn_entity(
     const tripoint_abs_ms &p, mtype_id id )
 {
-    std::unordered_map<tripoint_abs_ms, horde_entity>::iterator result;
+    std::optional<std::unordered_map<tripoint_abs_ms, horde_entity>::iterator> result;
     if( id.is_null() || !id.is_valid() ) {
         return result; // Bail out, blacklisted monster or something's wrong.
     }
@@ -151,7 +152,7 @@ std::unordered_map<tripoint_abs_ms, horde_entity>::iterator horde_map::spawn_ent
 }
 
 // TODO: check for a goal in horde_entity and put in active vs idle.
-std::unordered_map<tripoint_abs_ms, horde_entity>::iterator horde_map::spawn_entity(
+std::optional<std::unordered_map<tripoint_abs_ms, horde_entity>::iterator> horde_map::spawn_entity(
     const tripoint_abs_ms &p, const monster &mon )
 {
     std::unordered_map <tripoint_om_sm, std::unordered_map<tripoint_abs_ms, horde_entity>> &target_map =
@@ -159,7 +160,7 @@ std::unordered_map<tripoint_abs_ms, horde_entity>::iterator horde_map::spawn_ent
                 !is_alert( *mon.type ) ? immobile_monster_map :
                 ( mon.has_dest() || mon.wandf > 0 ) ? active_monster_map :
                 idle_monster_map;
-    std::unordered_map<tripoint_abs_ms, horde_entity>::iterator result;
+    std::optional<std::unordered_map<tripoint_abs_ms, horde_entity>::iterator> result;
     point_abs_om omp;
     tripoint_om_sm sm;
     std::tie( omp, sm ) = project_remain<coords::om>( project_to<coords::sm>( p ) );
@@ -167,10 +168,10 @@ std::unordered_map<tripoint_abs_ms, horde_entity>::iterator horde_map::spawn_ent
     // The [] operator creates a nested std::map if not present already.
     std::tie( result, inserted ) = target_map[sm].emplace( p, mon );
     if( inserted ) {
-        result->second.monster_data->set_pos_abs_only( p );
+        ( *result )->second.monster_data->set_pos_abs_only( p );
     } else {
         debugmsg( "Attempted to insert a %s at %s, but there's already a %s there!",
-                  mon.name(), p.to_string(), result->second.get_type()->nname() );
+                  mon.name(), p.to_string(), ( *result )->second.get_type()->nname() );
     }
     return result;
 }

--- a/src/horde_map.h
+++ b/src/horde_map.h
@@ -3,6 +3,7 @@
 #define CATA_SRC_HORDE_MAP_H
 
 #include <iterator>
+#include <optional>
 #include <unordered_map>
 #include <utility>
 #include <vector>
@@ -63,10 +64,11 @@ class horde_map
         std::vector<std::unordered_map<tripoint_abs_ms, horde_entity>*> entity_group_at(
             const tripoint_om_sm &p, int filter = horde_map_flavors::active | horde_map_flavors::idle |
                     horde_map_flavors::dormant | horde_map_flavors::immobile );
-        std::unordered_map<tripoint_abs_ms, horde_entity>::iterator
+        std::optional<std::unordered_map<tripoint_abs_ms, horde_entity>::iterator>
         spawn_entity( const tripoint_abs_ms &p, mtype_id id );
-        std::unordered_map<tripoint_abs_ms, horde_entity>::iterator spawn_entity( const tripoint_abs_ms &p,
-                const monster &mon );
+        std::optional<std::unordered_map<tripoint_abs_ms, horde_entity>::iterator> spawn_entity(
+            const tripoint_abs_ms &p,
+            const monster &mon );
         void signal_entities( const tripoint_abs_ms &origin, int volume );
         void insert( node_type &&node );
         void clear();

--- a/src/overmap.cpp
+++ b/src/overmap.cpp
@@ -1663,7 +1663,7 @@ void mongroup::wander( const overmap &om )
 
 horde_entity &overmap::spawn_monster( const tripoint_abs_ms &p, mtype_id id )
 {
-    return hordes.spawn_entity( p, id )->second;
+    return ( *hordes.spawn_entity( p, id ) )->second;
 }
 
 // Seeks through the submap looking for open areas.

--- a/src/savegame.cpp
+++ b/src/savegame.cpp
@@ -659,7 +659,7 @@ void overmap::unserialize( const JsonObject &jsobj )
             JsonArray monster_map_json = om_member;
             while( monster_map_json.has_more() ) {
                 tripoint_abs_ms monster_location;
-                std::unordered_map<tripoint_abs_ms, horde_entity>::iterator result;
+                std::optional<std::unordered_map<tripoint_abs_ms, horde_entity>::iterator> result;
                 monster_location.deserialize( monster_map_json.next_value() );
                 point_abs_om omp;
                 tripoint_om_sm monster_submap;
@@ -674,11 +674,11 @@ void overmap::unserialize( const JsonObject &jsobj )
                     result = hordes.spawn_entity( monster_location, new_monster );
                 }
 
-                if( result != std::unordered_map<tripoint_abs_ms, horde_entity>::iterator() ) {
-                    result->second.destination.deserialize( monster_map_json.next_value() );
-                    result->second.tracking_intensity = monster_map_json.next_int();
-                    result->second.last_processed.deserialize( monster_map_json.next_value() );
-                    result->second.moves = monster_map_json.next_int();
+                if( result.has_value() ) {
+                    ( *result )->second.destination.deserialize( monster_map_json.next_value() );
+                    ( *result )->second.tracking_intensity = monster_map_json.next_int();
+                    ( *result )->second.last_processed.deserialize( monster_map_json.next_value() );
+                    ( *result )->second.moves = monster_map_json.next_int();
                 } else {
                     // We deserialized something nasty, skip the rest of the stored values
                     monster_map_json.next_value();

